unitary/alpha/quantum_world.py

```diff
@@ -52,6 +52,16 @@ def add_object(self, obj: QuantumObject):
         obj.world = self
         obj.initial_effect()
 
+    def combine_with(self, other_world: "QuantumWorld"):
+        if self.used_object_keys.intersection(other_world.used_object_keys):
+            raise ValueError("Cannot combine two worlds with overlapping object keys")
+        self.objects.extend(other_world.objects)
+        self.used_object_keys.update(other_world.used_object_keys)
+        self.ancilla_names.update(other_world.ancilla_names)
+        self.post_selection.update(other_world.post_selection)
+        for op in other_world.circuit.all_operations():
+            self.circuit.append(op)
+
```

Comment:
Does `self.effect_history` need to be updated? Cleared maybe.

Reply:
It's now cleared.